### PR TITLE
Added axis parameter to tf.losses.softmax_cross_entropy. fixes #20866

### LINKED ISSUE #20866: Why tf.losses.softmax_cross_entropy doesn't have "dim" (axis) argument?
Hi!
The function "tf.losses.softmax_cross_entropy" calls "nn.softmax_cross_entropy_with_logits_v2" which has a dim argument with a default value of -1.
Can we add the same "dim" or "axis" parameter to the calling function " tf.losses.softmax_cross_entropy", so that we can choose which axis to apply the softmax  ?
Thanks!

### PR DESCRIPTION
**Pull request by coder3101**
The internal implementation of `tf.losses.softmax_cross_entropy`, uses `nn.softmax_cross_entropy_with_logits_v2` which has a dim argument with a default value of -1. The new argument `axis=-1` wraps the `dim` argument of `nn.softmax_cross_entropy_with_logits_v2`. Documentations updated too.

### CODE REVIEW

**Comment by coder3101** [comment]
Code formatting Issues. Will put another PR
